Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Get-ExchangeServerCertificates.ps1

```diff
@@ -61,10 +61,20 @@
                         $certStatus = ($cert.Status).ToString()
                     }
 
+                    if ([String]::IsNullOrEmpty($cert.SignatureAlgorithm.FriendlyName)) {
+                        $certSignatureAlgorithm = "Unknown"
+                        $certSignatureHashAlgorithm = "Unknown"
+                    } else {
+                        $certSignatureAlgorithm = $cert.SignatureAlgorithm.FriendlyName
+                        $certSignatureHashAlgorithm = ($cert.SignatureAlgorithm.FriendlyName | Select-String "(\w{3}\d{1,3})").Matches[0].Value
```

Comment:
Is this always going to work for all values that could be here? Might need to change this up just in case we don't have a `Select-String` that we find data on.

```
dpaul :: ~ » "bala" |Select-String "bah"
dpaul :: ~ » ("bala" |Select-String "bah").Matches[0].Value
InvalidOperation: Cannot index into a null array.
```

Reply:
Let me check if we can make this more bulletproof by doing OID mapping.

Reply:
Reworked. We now use a mapping to OID approach.